report/detailed_node.go

```diff
@@ -0,0 +1,54 @@
+package report
+
+import (
+	"reflect"
+	"strconv"
+)
+
+// MakeDetailedNode transforms a renderable node to a detailed node. It uses
+// aggregate metadata, plus the set of origin node IDs, to produce tables.
+func MakeDetailedNode(r Report, n RenderableNode) DetailedNode {
+	tables := []Table{}
+	{
+		rows := []Row{}
+		if val, ok := n.Metadata[KeyMaxConnCountTCP]; ok {
+			rows = append(rows, Row{"TCP connections", strconv.FormatInt(int64(val), 10), ""})
+		}
+		if val, ok := n.Metadata[KeyBytesIngress]; ok {
+			rows = append(rows, Row{"Bytes ingress", strconv.FormatInt(int64(val), 10), ""})
+		}
+		if val, ok := n.Metadata[KeyBytesEgress]; ok {
+			rows = append(rows, Row{"Bytes egress", strconv.FormatInt(int64(val), 10), ""})
+		}
+		if len(rows) > 0 {
+			tables = append(tables, Table{"Connections", true, rows})
+		}
+	}
+
+	// RenderableNode may be the result of merge operation(s), and so may have
+	// multiple origins. The ultimate goal here is to generate tables to view
+	// in the UI, so we skip the intermediate representations, but we could
+	// add them later.
+outer:
+	for _, id := range n.Origins {
+		table, ok := r.OriginTable(id)
+		if !ok {
+			continue
+		}
+		// Naïve equivalence-based deduplication.
+		for _, existing := range tables {
+			if reflect.DeepEqual(existing, table) {
```

Comment:
Yuk!  Why can you not compare node ids?  Or even better, does n.Origins offer set-like semantics?